### PR TITLE
beatrix: Add new test to verify add-on available behavior on new cata…

### PR DESCRIPTION
**Pull request by sbrossie**
…log version

### CODE REVIEW

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Override
+    protected KillbillConfigSource getConfigSource(final Map<String, String> extraProperties) {
+        final Map<String, String> allExtraProperties = new HashMap<String, String>(extraProperties);
+        allExtraProperties.put("org.killbill.catalog.uri", "catalogs/testCatalogAvailableAddOns");
+        return super.getConfigSource(null, allExtraProperties);
+    }
+
+    @Test(groups = "slow")
+    public void testXXX() throws Exception {
```
Comment (pierre) [inline beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java:53]: Better name? 😆

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        final UUID bpEntitlementId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec1), "externalKey", null, null, false, true, ImmutableList.<PluginProperty>of(), callContext);
+        assertListenerStatus();
+        Entitlement bpEntitlement = entitlementApi.getEntitlementForId(bpEntitlementId, callContext);
+
+        // Verify we cannot create an add-on Laser-Scope as it is not available in V1
+        final PlanPhaseSpecifier addOnSpec = new PlanPhaseSpecifier("Laser-Scope", BillingPeriod.MONTHLY, PriceListSet.DEFAULT_PRICELIST_NAME, null);
+        try {
+            entitlementApi.addEntitlement(bpEntitlement.getBundleId(), new DefaultEntitlementSpecifier(addOnSpec), clock.getUTCToday(), clock.getUTCToday(), false, ImmutableList.<PluginProperty>of(), callContext);
```
Comment (pierre) [inline beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java:75]: Missing `Assert.fail()`

**Review by pierre — CHANGES_REQUESTED** [review]

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java (hunk 1 above)
Comment (sbrossie) [reply beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java]: ack. fixed in https://github.com/killbill/killbill/pull/1421/commits/653f5c3e252ba30e4c3675f9d69aeb2c90e17898

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java (hunk 2 above)
Comment (sbrossie) [reply beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogAvailableAddOns.java:76]: ack. Fixed in https://github.com/killbill/killbill/pull/1421/commits/653f5c3e252ba30e4c3675f9d69aeb2c90e17898

**Pull request merged**